tiledb/sm/consolidator/consolidator.cc

```diff
@@ -210,50 +210,58 @@ void Consolidator::fragments_consolidate(
-      object_type(storage_manager->resources(), array_uri, &obj_type));
+  if (array_uri.is_tiledb()) {
+    throw_if_not_ok(
+        storage_manager->resources().rest_client()->post_consolidation_to_rest(
+            array_uri, config, &fragment_uris));
+  } else {
+    // Check if array exists
+    ObjectType obj_type;
+    throw_if_not_ok(
+        object_type(storage_manager->resources(), array_uri, &obj_type));
 
-  if (obj_type != ObjectType::ARRAY) {
-    throw ConsolidatorException(
-        "Cannot consolidate array; Array does not exist");
-  }
+    if (obj_type != ObjectType::ARRAY) {
+      throw ConsolidatorException(
+          "Cannot consolidate array; Array does not exist");
+    }
 
-  // Get encryption key from config
-  std::string encryption_key_from_cfg;
-  if (!encryption_key) {
-    bool found = false;
-    encryption_key_from_cfg = config.get("sm.encryption_key", &found);
-    assert(found);
-  }
+    // Get encryption key from config
+    std::string encryption_key_from_cfg;
+    if (!encryption_key) {
+      bool found = false;
+      encryption_key_from_cfg = config.get("sm.encryption_key", &found);
+      assert(found);
+    }
+
+    if (!encryption_key_from_cfg.empty()) {
+      encryption_key = encryption_key_from_cfg.c_str();
+      key_length = static_cast<uint32_t>(encryption_key_from_cfg.size());
+      std::string encryption_type_from_cfg;
+      bool found = false;
+      encryption_type_from_cfg = config.get("sm.encryption_type", &found);
+      assert(found);
+      auto [st, et] = encryption_type_enum(encryption_type_from_cfg);
+      throw_if_not_ok(st);
+      encryption_type = et.value();
 
-  if (!encryption_key_from_cfg.empty()) {
-    encryption_key = encryption_key_from_cfg.c_str();
-    key_length = static_cast<uint32_t>(encryption_key_from_cfg.size());
-    std::string encryption_type_from_cfg;
-    bool found = false;
-    encryption_type_from_cfg = config.get("sm.encryption_type", &found);
-    assert(found);
-    auto [st, et] = encryption_type_enum(encryption_type_from_cfg);
-    throw_if_not_ok(st);
-    encryption_type = et.value();
-
-    if (!EncryptionKey::is_valid_key_length(
-            encryption_type,
-            static_cast<uint32_t>(encryption_key_from_cfg.size()))) {
-      encryption_key = nullptr;
-      key_length = 0;
+      if (!EncryptionKey::is_valid_key_length(
+              encryption_type,
+              static_cast<uint32_t>(encryption_key_from_cfg.size()))) {
+        encryption_key = nullptr;
+        key_length = 0;
+      }
     }
-  }
 
-  // Consolidate
-  auto consolidator = Consolidator::create(
-      ConsolidationMode::FRAGMENT, config, storage_manager);
-  auto fragment_consolidator =
-      dynamic_cast<FragmentConsolidator*>(consolidator.get());
-  throw_if_not_ok(fragment_consolidator->consolidate_fragments(
-      array_name, encryption_type, encryption_key, key_length, fragment_uris));
+    // Consolidate
+    auto fragment_consolidator =
```

Comment:
removed `dynamic_cast` that should be avoided when possible.

Reply:
Let's not do that now. If we're going to do this, let's change all the places that use Consolidator::create for something better and consistent.

Reply:
Why shouldn't we do that immediate improvement? This is the only occurrence of a `dynamic_cast` in consolidation code, and the reason is that `consolidate_fragments` is a mehtod of `FragmentConsolidator` and not of its parent class `Consolidator`.

Reply:
After discussing offline, I am reverting this change, although I think it's improving the current situation, to get this long standing PR finally unblocked and merged.